CMakeLists.txt

```diff
@@ -77,95 +61,145 @@ elseif(${CMAKE_BUILD_TARGET} MATCHES "Both")
   set(BUILD_UVC_SHARED TRUE)
   set(BUILD_UVC_STATIC TRUE)
 else()
-  message( FATAL_ERROR "Invalid build type ${CMAKE_BUILD_TARGET}" )
+  message(FATAL_ERROR "Invalid build type ${CMAKE_BUILD_TARGET}" )
 endif()
 
 if(BUILD_UVC_SHARED)
   add_library(uvc SHARED ${SOURCES})
-  set_property(TARGET uvc PROPERTY VERSION ${libuvc_VERSION} )
-  set_property(TARGET uvc PROPERTY SOVERSION ${libuvc_VERSION_MAJOR} )
+  set_target_properties(uvc PROPERTIES
+    VERSION ${libuvc_VERSION}
+    SOVERSION ${libuvc_VERSION_MAJOR}
+    EXPORT_NAME UVCShared
+  )
   list(APPEND UVC_TARGETS uvc)
+  if(NOT LibUVC_STATIC)
+    add_library(LibUVC::UVC ALIAS uvc)
+  endif()
+  add_library(LibUVC::UVCShared ALIAS uvc)
 endif()
 
 if(BUILD_UVC_STATIC)
   add_library(uvc_static STATIC ${SOURCES})
-  set_target_properties(uvc_static PROPERTIES OUTPUT_NAME uvc)
+  set_target_properties(uvc_static PROPERTIES
+    OUTPUT_NAME uvc
+    EXPORT_NAME UVCStatic
+  )
   list(APPEND UVC_TARGETS uvc_static)
+  add_library(LibUVC::UVCStatic ALIAS uvc_static)
+  if(LibUVC_STATIC)
+    add_library(LibUVC::UVC ALIAS uvc_static)
+  endif()
 endif()
 
 configure_file(include/libuvc/libuvc_config.h.in
-  ${PROJECT_BINARY_DIR}/include/libuvc/libuvc_config.h @ONLY)
+  include/libuvc/libuvc_config.h
+  @ONLY
+)
 
-foreach(target_name ${UVC_TARGETS})
+foreach(target_name IN LISTS UVC_TARGETS)
+  target_include_directories(${target_name}
+    PUBLIC
+      $<INSTALL_INTERFACE:include>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include$<SEMICOLON>>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
```

Comment:
Different paths for includes for build and install phase supported via `INSTALL_INTERFACE` and `BUILD_INTERFACE` generator expressions.

Reply:
`$<SEMICOLON>` is redundant here. It used when I want to specify two include directories in single line.  
Original form: 
`$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include$<SEMICOLON>${CMAKE_CURRENT_BINARY_DIR}/include>`.
But it has no readability.